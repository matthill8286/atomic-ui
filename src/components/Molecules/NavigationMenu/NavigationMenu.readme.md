# Navigation Menu

NavigationMenu Items list, based on label and routing information

Uses `ActionLink` and `GroupedActionLinks` which has a handly utility for detecting the link type, either a href or a routing `to`
