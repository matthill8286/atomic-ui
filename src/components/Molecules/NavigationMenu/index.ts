export * from './NavigationMenu'
export * from './NavigationMenuList'
