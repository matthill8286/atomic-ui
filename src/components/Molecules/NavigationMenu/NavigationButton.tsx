import React, { FC } from 'react'
import { DropdownButton, DropdownButtonProps } from '@/components/Molecules/DropdownButton'
import { media, styled } from '@/styles'
import { IconUser as UserIcon } from '@excelwithbusiness/webmobile-svg-library'

export const StyledDropdownButton = styled(DropdownButton)`
  display: none;
  ${media.sm} {
    display: inline;
  }
`

export const NavigationButton: FC<DropdownButtonProps> = React.memo(
  ({ label, onClick, isOpen, additionalLabel, width, height, pictureSrc }) => {
    return (
      <StyledDropdownButton
        label={label}
        width={width || height}
        height={height}
        additionalLabel={additionalLabel}
        primaryIcon={<UserIcon height={40} width={40} />}
        isOpen={isOpen}
        onClick={onClick}
        pictureSrc={pictureSrc}
      />
    )
  }
)
