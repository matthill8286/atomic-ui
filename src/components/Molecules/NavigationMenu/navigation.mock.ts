import { QuickAction } from '@/components/Molecules/ActionLink'

export const navigationMockItems: QuickAction[] = [
  {
    actionLabel: 'Profile',
    to: '/profile',
    links: [],
  },
  {
    actionLabel: 'Comments',
    to: '/comments',
    links: [],
  },
  {
    actionLabel: 'Goals',
    to: '/goals',
    links: [],
  },
  {
    actionLabel: 'Progress',
    to: '/profile/progress',
    links: [],
  },
]
