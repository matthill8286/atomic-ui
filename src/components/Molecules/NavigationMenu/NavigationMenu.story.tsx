import { action } from '@storybook/addon-actions'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { navigationMockItems } from './navigation.mock'
import { NavigationMenu } from './NavigationMenu'
import Readme from './NavigationMenu.readme.md'

storiesOf('Design System/Molecules/NavigationMenu', module)
  .add(
    'Default',
    () => (
      <div style={{ display: 'flex', width: '100%', justifyContent: 'center' }}>
        <NavigationMenu
          button={{
            actionLabel: 'Log Out',
            href: '/test',
            onClick: action('logout clicked'),
          }}
          iconAsMainUi
          initialLink={undefined}
          navActionList={navigationMockItems}
          extendedNavigationHandler={action('item clicked')}
        />
      </div>
    ),
    {
      info: Readme,
    }
  )
  .add(
    'With text as initial link',
    () => (
      <div style={{ display: 'flex', width: '100%', justifyContent: 'center' }}>
        <NavigationMenu
          button={{
            actionLabel: 'Log Out',
            href: '/test',
            onClick: action('logout clicked'),
          }}
          iconAsMainUi={false}
          initialLink="Events"
          navActionList={navigationMockItems}
          extendedNavigationHandler={action('item clicked')}
        />
      </div>
    ),
    {
      info: Readme,
    }
  )
