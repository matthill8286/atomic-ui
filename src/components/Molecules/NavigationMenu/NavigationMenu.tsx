import React, { FC, useEffect, useRef } from 'react'
import { useDropdownState } from '@/components/Helper/useDropdownState'
import {
  StyledContainer,
  StyledDropdown,
} from '@/components/Molecules/DropdownButton/DropdownButton'
import { NavigationButton } from './NavigationButton'
import {
  NavigationButtonType,
  NavigationMenuList,
  NavigationMenuListProps,
} from './NavigationMenuList'
import { CopyText } from '@/components/Atoms/Typography'
import { breakpoints, styled } from '@/styles'
import { ThemeColors } from '@/types'
import { useWindowDimensions } from '@/components/Helper'

export interface NavigationMenuProps {
  className?: string
  index?: number
  button?: NavigationButtonType | undefined
  testIdSuffix?: string
  iconAsMainUi?: boolean
  initialLink?: string
  pictureSrc?: string
  copyColor?: ThemeColors
  navActionList: NavigationMenuListProps['navActionList']
  extendedNavigationHandler?: (item?: NavigationMenuListProps) => void
}

export const StyledCopy = styled(CopyText)`
  ${({ theme }) => `
    display: flex;
    width: 100%;

    :hover {
        cursor: pointer;
    }
  `}
`

export const NavigationMenu: FC<NavigationMenuProps> = ({
  className,
  navActionList = [],
  button,
  testIdSuffix,
  extendedNavigationHandler,
  iconAsMainUi = true,
  initialLink,
  copyColor,
  pictureSrc,
}) => {
  const { element, toggleDropdown, showDropdown } = useDropdownState<HTMLDivElement>()
  const { breakpoint: currentBreakpoint } = useWindowDimensions()
  const isMobile = currentBreakpoint < breakpoints.md

  const ref = useRef<HTMLElement>(null)

  const onToggleDropdown = () => {
    toggleDropdown()
  }

  const handleMenuItemClick = item => {
    if (showDropdown) {
      toggleDropdown()
    }

    if (extendedNavigationHandler) {
      extendedNavigationHandler(item)
    }
  }

  const handleClickOutside = (event: Event): void => {
    if (!ref.current?.contains(event.target as Node) && showDropdown) {
      toggleDropdown()
    }
  }

  useEffect(() => {
    document.addEventListener('click', handleClickOutside)

    return () => {
      document.removeEventListener('click', handleClickOutside)
    }
  })

  const myNavigationContent = (
    <NavigationMenuList
      navActionList={navActionList}
      actionButton={button}
      flexed
      padding="sm"
      onClick={handleMenuItemClick}
    />
  )

  return (
    <StyledContainer
      className={className}
      ref={element}
      data-test={`navigation-dropdown-${testIdSuffix}`}>
      {iconAsMainUi ? (
        <NavigationButton
          height={40}
          width={40}
          data-test="ewb-app-header-navigation-button"
          onClick={onToggleDropdown}
          isOpen={showDropdown}
          pictureSrc={pictureSrc}
        />
      ) : (
        <StyledCopy
          color={copyColor}
          fontSize="sm"
          margin="auto"
          data-test="ewb-header-navigation-link"
          onClick={onToggleDropdown}>
          {initialLink}
        </StyledCopy>
      )}
      <StyledDropdown isMobile={isMobile} surface="white" showDropdown={showDropdown}>
        {myNavigationContent}
      </StyledDropdown>
    </StyledContainer>
  )
}
