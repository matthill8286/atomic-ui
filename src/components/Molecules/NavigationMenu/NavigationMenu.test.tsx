import React from 'react'
import { renderWithThemeAndRouter } from '@/testRenderer'
import { navigationMockItems } from './navigation.mock'
import { NavigationMenu } from './NavigationMenu'

describe('NavigationMenu', () => {
  it('should render with items', () => {
    const wrapper = renderWithThemeAndRouter(
      <NavigationMenu
        extendedNavigationHandler={() => jest.fn()}
        navActionList={navigationMockItems}
        testIdSuffix="data-test-menu"
        initialLink={undefined}
        iconAsMainUi
        button={{ actionLabel: 'Log Out', onClick: () => jest.fn(), href: '/letuslogout' }}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with text for the main actiuon', () => {
    const wrapper = renderWithThemeAndRouter(
      <NavigationMenu
        extendedNavigationHandler={() => jest.fn()}
        navActionList={navigationMockItems}
        testIdSuffix="data-test-menu"
        initialLink="Events"
        iconAsMainUi={false}
        button={{ actionLabel: 'Log Out', onClick: () => jest.fn(), href: '/letuslogout' }}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
