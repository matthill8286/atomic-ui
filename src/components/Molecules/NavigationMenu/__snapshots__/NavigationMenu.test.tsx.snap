// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu should render with items 1`] = `
.c5 {
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c16 {
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c12 {
  display: block;
  box-sizing: border-box;
  line-height: 1.43;
  overflow: hidden;
  box-shadow: 0 3px 4px 0 rgba(0,0,0,0.07),0 3px 3px -2px rgba(0,0,0,0.06),0 1px 8px 0 rgba(0,0,0,0.10);
  color: #000000;
  background-color: #FFFFFF;
  border: none;
  border-radius: 4px;
  height: 100%;
  width: 100%;
}

.c10 {
  position: relative;
  outline: none;
  width: 100%;
}

.c10 .Cardstyled__StyledBadgeLineWrapper-sc-1gxmhl7-0 {
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

.c8 svg {
  width: 40px;
  height: 40px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #666666;
}

.c8 svg g,
.c8 svg p {
  fill: #666666;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  border: 2px solid #FFFFFF;
  background: #FFFFFF;
  color: #666666;
}

.c1 .c7 {
  padding: 0 8px;
}

.c1 .Status-y46az8-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c1:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c1:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c1 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

.c0 {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  position: relative;
}

.c11 {
  position: absolute;
  left: -100px;
  top: 56px;
  width: 280px;
  margin: 16px 0 0 0;
  opacity: 0;
  visibility: hidden;
  z-index: 99;
  -webkit-transform: translateY(-16px);
  -ms-transform: translateY(-16px);
  transform: translateY(-16px);
  -webkit-transition: visibility 0.15s ease-in-out,opacity 0.15s ease-in-out,-webkit-transform 0.15s ease-in-out;
  -webkit-transition: visibility 0.15s ease-in-out,opacity 0.15s ease-in-out,transform 0.15s ease-in-out;
  transition: visibility 0.15s ease-in-out,opacity 0.15s ease-in-out,transform 0.15s ease-in-out;
}

.c2 {
  padding: 0 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border-color: transparent;
  border-radius: 4px;
  cursor: pointer;
}

.c2:active,
.c2:focus {
  outline: none;
}

.c2:hover {
  box-shadow: none;
}

.c2 .c7 {
  padding: 0;
}

.c2::-moz-focus-inner {
  border: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c6 {
  justify-self: start;
  white-space: nowrap;
}

.c9 {
  margin: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c3 {
  display: none;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #999999;
  opacity: 1.0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  position: relative;
  padding: 16px;
}

.c18::after {
  content: '';
  position: absolute;
  display: none;
  bottom: -2px;
  height: 1px;
  background-color: #666666;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
  -webkit-transition: -webkit-transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  -webkit-transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
}

.c18:hover::after {
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #999999;
  opacity: 1.0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  position: relative;
  padding: 16px;
}

.c15::after {
  content: '';
  position: absolute;
  display: none;
  bottom: -2px;
  height: 1px;
  background-color: #666666;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
  -webkit-transition: -webkit-transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  -webkit-transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
}

.c15:hover::after {
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c17 {
  display: block;
  color: #666666;
}

.c13 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 0;
  margin-right: 0;
}

.c13:hover {
  cursor: pointer;
  background: #E5E5E5;
}

.c14 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c14 > a {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  color: #333333;
  border-top: 1px solid #E5E5E5;
}

.c19:hover {
  cursor: pointer;
  background: #E5E5E5;
}

@media (min-width:512px) {
  .c10:hover .Cardstyled__StyledBadgeLineWrapper-sc-1gxmhl7-0,
  .c10:hover .Cardstyled__StyledFeaturedAssetsContainer-sc-1gxmhl7-3 {
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c2 {
    width: 130px;
  }
}

@media (min-width:512px) {
  .c3 {
    display: inline;
  }
}

<div
  className="c0"
  data-test="navigation-dropdown-data-test-menu"
>
  <button
    className="c1 c2 c3"
    disabled={false}
    onClick={[Function]}
    size="md"
    type="button"
  >
    <div
      className="c4"
    >
      <span
        className="c5 c6"
        color="grey5"
        fontSize="sm"
      />
      <div
        className="c7 c8 c9"
        color="grey5"
        height={40}
        width={40}
      >
        <svg
          height={40}
          viewBox="0 0 480 480"
          width={40}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M240 0C107.664 0 0 107.664 0 240c0 57.96 20.656 111.184 54.992 152.704.088.12.096.272.192.384 24.792 29.896 55.928 52.816 90.624 67.624.4.168.792.352 1.192.52 2.808 1.184 5.648 2.28 8.496 3.352 1.12.424 2.24.856 3.376 1.264 2.456.88 4.928 1.712 7.416 2.512 1.592.512 3.184 1.016 4.792 1.496 2.2.656 4.408 1.288 6.632 1.888 1.952.528 3.92 1.016 5.888 1.488 1.992.48 3.992.96 6 1.384 2.24.48 4.504.904 6.776 1.32 1.824.336 3.64.688 5.48.984 2.52.408 5.056.728 7.6 1.056 1.64.208 3.272.448 4.92.624 2.88.304 5.784.52 8.696.72 1.352.096 2.696.24 4.056.312 4.248.24 8.544.368 12.872.368s8.624-.128 12.888-.352c1.36-.072 2.704-.216 4.056-.312 2.912-.208 5.816-.416 8.696-.72 1.648-.176 3.28-.416 4.92-.624 2.544-.328 5.08-.648 7.6-1.056 1.832-.296 3.656-.648 5.48-.984 2.264-.416 4.528-.84 6.776-1.32 2.008-.432 4-.904 6-1.384 1.968-.48 3.936-.968 5.888-1.488a232.661 232.661 0 0011.424-3.384c2.488-.8 4.96-1.632 7.416-2.512 1.128-.408 2.248-.84 3.376-1.264a243.168 243.168 0 008.496-3.352c.4-.168.792-.352 1.192-.52a239.997 239.997 0 0090.624-67.624c.096-.112.104-.272.192-.384C459.344 351.184 480 297.96 480 240 480 107.664 372.336 0 240 0zm97.256 441.76c-.12.056-.232.12-.352.176-2.856 1.376-5.76 2.672-8.688 3.936-.664.28-1.32.568-1.984.848a217.023 217.023 0 01-7.76 3.064 159.85 159.85 0 01-3.272 1.192 230.572 230.572 0 01-6.976 2.368c-1.456.464-2.92.904-4.384 1.336a202.08 202.08 0 01-6.28 1.784c-1.776.472-3.568.904-5.36 1.328-1.88.448-3.752.904-5.648 1.304-2.072.44-4.16.816-6.24 1.192-1.688.312-3.368.64-5.072.912-2.344.368-4.712.664-7.072.96-1.496.192-2.984.416-4.496.576-2.696.288-5.416.472-8.128.664-1.208.08-2.408.216-3.632.28-3.96.208-7.928.32-11.912.32s-7.952-.112-11.904-.32c-1.216-.064-2.416-.192-3.632-.28-2.72-.184-5.432-.376-8.128-.664-1.512-.16-3-.384-4.496-.576-2.36-.296-4.728-.592-7.072-.96-1.704-.272-3.384-.6-5.072-.912-2.088-.376-4.176-.76-6.24-1.192-1.896-.4-3.776-.856-5.648-1.304-1.792-.432-3.584-.856-5.36-1.328-2.104-.56-4.2-1.168-6.28-1.784a212.543 212.543 0 01-4.384-1.336 223.256 223.256 0 01-6.976-2.368 217.023 217.023 0 01-11.032-4.256c-.664-.272-1.312-.56-1.976-.84a223.812 223.812 0 01-8.696-3.936 7.216 7.216 0 01-.352-.176c-27.912-13.504-52.568-32.672-72.576-55.952 15.464-56.944 59.24-102.848 115.56-121.112 1.112.68 2.272 1.288 3.416 1.928.672.376 1.336.776 2.016 1.136a99.592 99.592 0 007.272 3.512c1.896.832 3.856 1.536 5.808 2.256.384.136.768.288 1.152.424 10.848 3.84 22.456 6.04 34.6 6.04s23.752-2.2 34.592-6.04c.384-.136.768-.288 1.152-.424 1.952-.72 3.912-1.424 5.808-2.256a99.592 99.592 0 007.272-3.512c.68-.368 1.344-.76 2.016-1.136 1.144-.64 2.312-1.248 3.432-1.936 56.32 18.272 100.088 64.176 115.56 121.112-20.008 23.272-44.664 42.44-72.576 55.952zM152 176c0-48.52 39.48-88 88-88s88 39.48 88 88c0 30.864-16.008 58.024-40.128 73.736a84.742 84.742 0 01-9.8 5.48c-.4.192-.792.392-1.192.576-23.168 10.536-50.592 10.536-73.76 0-.4-.184-.8-.384-1.192-.576a84.083 84.083 0 01-9.8-5.48C168.008 234.024 152 206.864 152 176zm269.832 194.584c-18.136-53.552-59.512-96.832-112.376-117.392C330.6 234.144 344 206.64 344 176c0-57.344-46.656-104-104-104s-104 46.656-104 104c0 30.64 13.4 58.144 34.552 77.192-52.864 20.568-94.24 63.84-112.376 117.392C31.672 333.792 16 288.704 16 240 16 116.488 116.488 16 240 16s224 100.488 224 224c0 48.704-15.672 93.792-42.168 130.584z"
          />
        </svg>
      </div>
    </div>
  </button>
  <div
    className="c10 c11"
  >
    <div
      className="c12"
      display="block"
      elevation={2}
      shape="rounded-small"
    >
      <div
        onClick={[Function]}
      >
        <div
          className="c13"
        >
          <div
            className="c14"
            data-test="grouped-action-links-undefined"
          >
            <a
              className="c15"
              href="/profile"
              onClick={[Function]}
            >
              <span
                className="c16 c17"
                color="grey5"
                fontSize="sm"
              >
                Profile
              </span>
            </a>
          </div>
        </div>
        <div
          className="c13"
        >
          <div
            className="c14"
            data-test="grouped-action-links-undefined"
          >
            <a
              className="c15"
              href="/comments"
              onClick={[Function]}
            >
              <span
                className="c16 c17"
                color="grey5"
                fontSize="sm"
              >
                Comments
              </span>
            </a>
          </div>
        </div>
        <div
          className="c13"
        >
          <div
            className="c14"
            data-test="grouped-action-links-undefined"
          >
            <a
              className="c15"
              href="/goals"
              onClick={[Function]}
            >
              <span
                className="c16 c17"
                color="grey5"
                fontSize="sm"
              >
                Goals
              </span>
            </a>
          </div>
        </div>
        <div
          className="c13"
        >
          <div
            className="c14"
            data-test="grouped-action-links-undefined"
          >
            <a
              className="c15"
              href="/profile/progress"
              onClick={[Function]}
            >
              <span
                className="c16 c17"
                color="grey5"
                fontSize="sm"
              >
                Progress
              </span>
            </a>
          </div>
        </div>
        <a
          className="c18 c19"
          color="grey5"
          href="/letuslogout"
          onClick={[Function]}
          target="_self"
        >
          <span
            className="c16 c17"
            color="grey5"
            fontSize="sm"
          >
            Log Out
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavigationMenu should render with text for the main actiuon 1`] = `
.c1 {
  margin: auto;
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c9 {
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c5 {
  display: block;
  box-sizing: border-box;
  line-height: 1.43;
  overflow: hidden;
  box-shadow: 0 3px 4px 0 rgba(0,0,0,0.07),0 3px 3px -2px rgba(0,0,0,0.06),0 1px 8px 0 rgba(0,0,0,0.10);
  color: #000000;
  background-color: #FFFFFF;
  border: none;
  border-radius: 4px;
  height: 100%;
  width: 100%;
}

.c3 {
  position: relative;
  outline: none;
  width: 100%;
}

.c3 .Cardstyled__StyledBadgeLineWrapper-sc-1gxmhl7-0 {
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c0 {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  position: relative;
}

.c4 {
  position: absolute;
  left: -100px;
  top: 56px;
  width: 280px;
  margin: 16px 0 0 0;
  opacity: 0;
  visibility: hidden;
  z-index: 99;
  -webkit-transform: translateY(-16px);
  -ms-transform: translateY(-16px);
  transform: translateY(-16px);
  -webkit-transition: visibility 0.15s ease-in-out,opacity 0.15s ease-in-out,-webkit-transform 0.15s ease-in-out;
  -webkit-transition: visibility 0.15s ease-in-out,opacity 0.15s ease-in-out,transform 0.15s ease-in-out;
  transition: visibility 0.15s ease-in-out,opacity 0.15s ease-in-out,transform 0.15s ease-in-out;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #999999;
  opacity: 1.0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  position: relative;
  padding: 16px;
}

.c11::after {
  content: '';
  position: absolute;
  display: none;
  bottom: -2px;
  height: 1px;
  background-color: #666666;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
  -webkit-transition: -webkit-transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  -webkit-transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
}

.c11:hover::after {
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #999999;
  opacity: 1.0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  position: relative;
  padding: 16px;
}

.c8::after {
  content: '';
  position: absolute;
  display: none;
  bottom: -2px;
  height: 1px;
  background-color: #666666;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
  -webkit-transition: -webkit-transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  -webkit-transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
}

.c8:hover::after {
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c10 {
  display: block;
  color: #666666;
}

.c6 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 0;
  margin-right: 0;
}

.c6:hover {
  cursor: pointer;
  background: #E5E5E5;
}

.c7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c7 > a {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  color: #333333;
  border-top: 1px solid #E5E5E5;
}

.c12:hover {
  cursor: pointer;
  background: #E5E5E5;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c2:hover {
  cursor: pointer;
}

@media (min-width:512px) {
  .c3:hover .Cardstyled__StyledBadgeLineWrapper-sc-1gxmhl7-0,
  .c3:hover .Cardstyled__StyledFeaturedAssetsContainer-sc-1gxmhl7-3 {
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@media (min-width:512px) {

}

<div
  className="c0"
  data-test="navigation-dropdown-data-test-menu"
>
  <p
    className="c1 c2"
    color="grey5"
    data-test="ewb-header-navigation-link"
    fontSize="sm"
    onClick={[Function]}
  >
    Events
  </p>
  <div
    className="c3 c4"
  >
    <div
      className="c5"
      display="block"
      elevation={2}
      shape="rounded-small"
    >
      <div
        onClick={[Function]}
      >
        <div
          className="c6"
        >
          <div
            className="c7"
            data-test="grouped-action-links-undefined"
          >
            <a
              className="c8"
              href="/profile"
              onClick={[Function]}
            >
              <span
                className="c9 c10"
                color="grey5"
                fontSize="sm"
              >
                Profile
              </span>
            </a>
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            data-test="grouped-action-links-undefined"
          >
            <a
              className="c8"
              href="/comments"
              onClick={[Function]}
            >
              <span
                className="c9 c10"
                color="grey5"
                fontSize="sm"
              >
                Comments
              </span>
            </a>
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            data-test="grouped-action-links-undefined"
          >
            <a
              className="c8"
              href="/goals"
              onClick={[Function]}
            >
              <span
                className="c9 c10"
                color="grey5"
                fontSize="sm"
              >
                Goals
              </span>
            </a>
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            data-test="grouped-action-links-undefined"
          >
            <a
              className="c8"
              href="/profile/progress"
              onClick={[Function]}
            >
              <span
                className="c9 c10"
                color="grey5"
                fontSize="sm"
              >
                Progress
              </span>
            </a>
          </div>
        </div>
        <a
          className="c11 c12"
          color="grey5"
          href="/letuslogout"
          onClick={[Function]}
          target="_self"
        >
          <span
            className="c9 c10"
            color="grey5"
            fontSize="sm"
          >
            Log Out
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
